src/pages/home/sidebar/SidebarLinks.tsx

```diff
@@ -68,7 +68,16 @@ function SidebarLinks({insets, optionListItems, isLoading, priorityMode = CONST.
             // or when continuously clicking different LHNs, only apply to small screen
             // since getTopmostReportId always returns on other devices
             const reportActionID = Navigation.getTopmostReportActionId();
-            if ((option.reportID === Navigation.getTopmostReportId() && !reportActionID) || (shouldUseNarrowLayout && isActiveReport(option.reportID) && !reportActionID)) {
+
+            // Prevent opening a new Report page if the user quickly taps on another conversation
+            // before the first one is displayed.
+            const shouldOpenReportInNarrowLayout = Navigation.getActiveRoute() !== '/home' && shouldUseNarrowLayout;
```

Comment:
@huult Sorry for the previous confusion, but this var name I suggested actually contradicts what we do here. I'm not great at naming variables, but it should be something that clearly defines its purpose. I'm not sure what the best would be, but `shouldBlockReportNavigation` comes to mind. What do you think, @yuwenmemon?